apps/prairielearn/src/lib/assessment.js

```diff
@@ -36,486 +38,414 @@ const InstanceLogSchema = z.object({
-   *
-   * @param {string} assessment_instance_id - The assessment instance to grade.
-   * @param {string} authn_user_id - The current authenticated user.
-   * @param {function} callback - A callback(err, updated) function.
-   */
-  update(assessment_instance_id, authn_user_id, callback) {
-    debug('update()');
-    let updated;
-    async.series(
-      [
-        async () => {
-          await sqldb.runInTransactionAsync(async () => {
-            await sqldb.callAsync('assessment_instances_lock', [assessment_instance_id]);
-            const updateResult = await sqldb.callOneRowAsync('assessment_instances_update', [
-              assessment_instance_id,
-              authn_user_id,
-            ]);
-            updated = updateResult.rows[0].updated;
-            if (!updated) return null; // skip if not updated
-
-            // if updated, regrade to pick up max_points changes, etc.
-            await sqldb.callOneRowAsync('assessment_instances_grade', [
-              assessment_instance_id,
-              authn_user_id,
-              null, // credit
-              true, // only_log_if_score_updated
-            ]);
-          });
-        },
-        (callback) => {
-          // Don't try to update LTI score if the assessment wasn't updated.
-          if (!updated) return callback(null);
-
-          // NOTE: It's important that this is run outside of `runInTransaction`
-          // above. This will hit the network, and as a rule we don't do any
-          // potentially long-running work inside of a transaction.
-          ltiOutcomes.updateScore(assessment_instance_id, (err) => {
-            if (ERR(err, callback)) return;
-            callback(null);
-          });
-        },
-      ],
-      (err) => {
-        if (ERR(err, callback)) return;
-        callback(null, updated);
-      },
-    );
-  },
-
-  /**
-   * Grade all questions in an assessment instance and (optionally) close it.
-   *
-   * All user-facing routes should set `requireOpen` to true. However, internal
-   * functions that asynchronously grade exams can set `requireOpen` to false
-   * if needed.
-   *
-   * @param {number} assessment_instance_id - The assessment instance to grade.
-   * @param {string | null} authn_user_id - The current authenticated user.
-   * @param {boolean} requireOpen - Whether to enforce that the assessment instance is open before grading.
-   * @param {boolean} close - Whether to close the assessment instance after grading.
-   * @param {boolean} overrideGradeRate - Whether to override grade rate limits.
-   * @param {function} callback - A callback(err) function.
-   */
-  gradeAssessmentInstance(
+  ]);
+  return result.rows[0].assessment_instance_id;
+}
+
+/**
+ * Add new questions to the assessment instance and regrade it if necessary.
+ *
+ * @param {string} assessment_instance_id - The assessment instance to grade.
+ * @param {string} authn_user_id - The current authenticated user.
```

Comment:
Consider adding an `@returns` JSDoc here to document that this returns a boolean that indicates if the assessment instance was updated or not.

Reply:
Done in fa84191bfbb4c81d5657adec0cc55479cfbc5c05. I also added this in all functions.